modules/openapi-generator/src/main/resources/swift5/APIs.mustache

```diff
@@ -37,17 +37,19 @@ import Vapor
     {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} let method: String
     {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} let URLString: String
     {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} let requestTask: RequestTask = RequestTask()
+    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} let isRequestAuthenticated: Bool
```

Comment:
Hey @Jonas1893, thanks for creating this PR.
I like this approach.
The only thing that I don't like is the name `isRequestAuthenticated` because I think is misleading.
It gives the impression that the request is already authenticated.
What about `requiresAuthentication`?

Reply:
Hi @4brunu yea thats better. I also tried to improve the samples for URLSession a bit, seeing as the bearer token was previously added as a static header to the generator. This could lead to undesired behavior because of race conditions if multiple requests are executed in parallel, some with and some without token and maybe also a token refresh updates the token in-between.

Reply:
Hi @Jonas1893.
Looks good overall.
The issue I see in the change of the samples for URLSession is that you can't fetch a new token from the server if you need it, and in the method `execute` you can get a new token from the server if you need it.
So I'm not sure about that change.
CI if failing, can you please take a look at it?
Thanks

Reply:
True, I fixed the samples to dynamically inject a new fetched token into the request headers. 

I don't think the error from CI is caused by my change
`[ERROR] Plugin org.apache.maven.plugins:maven-shade-plugin:3.2.0 or one of its dependencies could not be resolved:`

Looks like there is a problem fetching one of the mvn dependencies